src/main/native/org/apache/commons/crypto/random/OpenSslCryptoRandomNative.c

```diff
@@ -294,49 +278,57 @@ static unsigned long pthreads_thread_id(void)
  * If using an Intel chipset with RDRAND, the high-performance hardware
  * random number generator will be used.
  */
-static ENGINE * openssl_rand_init(void)
+static ENGINE * openssl_rand_init(JNIEnv *env)
 {
-  locks_setup();
+  if (OPENSSL_VERSION_NUMBER < VERSION_1_1_X) {
```

Comment:
Looks like another compile-time check, if `OPENSSL_VERSION_NUMBER` really comes from the openssl headers.

Reply:
Done.